src/libopensc/pkcs15-gemsafeGPK.c

```diff
@@ -283,7 +283,7 @@ static int sc_pkcs15emu_gemsafeGPK_init(sc_pkcs15_card_t *p15card)
 		}
 
 		kinfo[num_keyinfo].fileid = i;
-		sc_pkcs15_format_id("NONE", &kinfo[num_keyinfo].id); 
+		sc_pkcs15_format_id("FF", &kinfo[num_keyinfo].id); 
```

Comment:
Why did you pick "FF". I think the 0.15.0 current code would would not change the id, which may be uninitialized 
A beter choice might be "", that should cause sc_hex_to_bin to set id->len=0;
But  Later in line 416, the cert_info.id is copied to the kinfo[n].id so it get overwritten anyway. 

(I wrote this code in 2005, but don't have any of these cards. I doubt the the code is even used by anyone.)